docs/proposals/20191016-e2e-test-framework.md

```diff
@@ -0,0 +1,207 @@
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+    - [As a Cluster API provider implementor I want to know my provider behaves in a way that Cluster API expects](#as-a-cluster-api-provider-implementor-i-want-to-know-my-provider-behaves-in-a-way-that-cluster-api-expects)
+    - [As a developer in the Cluster API ecosystem I want to have confidence that my change does not break expected behaviors](#as-a-developer-in-the-cluster-api-ecosystem-i-want-to-have-confidence-that-my-change-does-not-break-expected-behaviors)
+    - [As a Cluster API developer, I want to be sure I’m not accidentally disrupting existing providers.](#as-a-cluster-api-developer-i-want-to-be-sure-im-not-accidentally-disrupting-existing-providers)
+    - [As the release manager of a Cluster API provider, I want to be able to recommend new releases to users as soon as they come up](#as-the-release-manager-of-a-cluster-api-provider-i-want-to-be-able-to-recommend-new-releases-to-users-as-soon-as-they-come-up)
+    - [As a user of Cluster API, I want to be sure new releases I’m installing won’t cause regressions.](#as-a-user-of-cluster-api-i-want-to-be-sure-new-releases-im-installing-wont-cause-regressions)
+  - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+    - [Behaviors to test](#behaviors-to-test)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Alternatives](#alternatives)
+- [Upgrade Strategy](#upgrade-strategy)
+- [Additional Details](#additional-details)
+  - [Test Plan](#test-plan)
+  - [Version Skew Strategy [optional]](#version-skew-strategy-optional)
+- [Implementation History](#implementation-history)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Glossary
+
+Refer to the [Cluster API Book Glossary](https://cluster-api.sigs.k8s.io/reference/glossary.html).
+
+If this proposal adds new terms, or defines some, make the changes to the book's glossary when in PR stage.
+
+Basic e2es:
+The Reconcilers are exercised by creating a cluster and deleting a cluster
+Comprehensive e2es:
+Many of the components and behaviors of Cluster API are exercised.
+Creates a number of clusters and tears them down.
+
+## Summary
+
+Cluster API's providers could benefit from having a set of generic behavioral e2e tests. Most providers have some form
+of end-to-end testing but they are not uniform and do not all test the same behavior that Cluster API promises.
+
+A pluggable set of behavioral tests for Cluster API Providers would prove beneficial for the project health as a whole.
+
+As stated in the non-goals, this proposal does not intend to define the conformance requirements for Cluster API providers.
+
+## Motivation
+
+Every infrastructure and bootstrap provider wants to know if it "works", but testing today is a manual process.
+Developers test one or two well-trod scenarios if testing at all before merging PRs. This test suite, when properly
```

Comment:
Can this be simplified as:
```
Developers want to test more than one or two well-trod scenarios on PRs in a reasonable time while not compromising on the quality signals.
```

Reply:
I don't want to make guarantees on the amount of time the suite will take to run as that's entirely up to to the provider. But I do think this paragraph could use simplification. I'll ping you on update.